packages/core/src/amazonq/extApi.ts

```diff
@@ -0,0 +1,31 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import vscode from 'vscode'
+import { VSCODE_EXTENSION_ID } from '../shared/utilities'
+import { ToolkitError } from '../shared'
+import { GenerateAssistantResponseCommandOutput, GenerateAssistantResponseRequest } from '@amzn/codewhisperer-streaming'
+import { FeatureAuthState } from '../codewhisperer/util/authUtil'
+
+export interface qAPI {
+    chatApi: {
+        chat(request: GenerateAssistantResponseRequest): Promise<GenerateAssistantResponseCommandOutput>
+    }
+    authApi: {
+        reauthIfNeeded(): Promise<void>
+        getChatAuthState(): Promise<FeatureAuthState>
+    }
+}
+
+/**
+ * Get the extension API for Q
+ */
+export async function getQAPI(): Promise<qAPI> {
+    const ext = vscode.extensions.getExtension(VSCODE_EXTENSION_ID.amazonq)
+    if (!ext) {
+        throw new ToolkitError('Amazon Q is not installed', { code: 'AmazonQNotInstalled' })
```

Comment:
Can we avoid this? Application Composer's integration with Q should be optional. Composer should still be able to open when Q is not installed (though the Q integration will be unavailable).